openapi3/issue796_test.go

```diff
@@ -0,0 +1,18 @@
+package openapi3
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIssue796(t *testing.T) {
+	// Need to set CircularReferenceCounter to > 10
+	CircularReferenceCounter = 20
```

Comment:
To pass CI you'll need to edit the line in `go.yml` to be 
```shell
[[ $(git grep -InE 'return .*validateExtensions' -- openapi3 | wc -l) -eq $(git grep -InE '^\s+Extensions.+`' -- openapi3 | wc -l) ]]
```
Note the first regexp and flags is different from the one in https://github.com/getkin/kin-openapi/blob/fd52f34d334769c4cbcabba87272a2c2a5b7aa4f/.github/workflows/go.yml#L129

Here I feel you'll need to do a `defer` trick like https://github.com/getkin/kin-openapi/blob/fd52f34d334769c4cbcabba87272a2c2a5b7aa4f/openapi3/issue615_test.go#L23-L25

Reply:
Thanks. Done.